remote-trait-object/src/ipc/multiplex.rs

```diff
@@ -14,73 +14,114 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
+use crate::ipc::{IpcRecv, IpcSend, RecvError, Terminate};
 use crossbeam::channel::{self, Receiver, Sender};
+use parking_lot::Mutex;
 use std::thread;
 
 pub struct MultiplexResult {
     pub request_recv: Receiver<String>,
     pub response_recv: Receiver<String>,
+    pub multiplexed_send: Sender<String>,
     pub multiplexer: Multiplexer,
 }
 
 pub struct Multiplexer {
     receiver_thread: Option<thread::JoinHandle<()>>,
-    termination_send: Sender<()>,
+    receiver_terminator: Option<Mutex<Box<dyn Terminate>>>,
+    sender_thread: Option<thread::JoinHandle<()>>,
+    sender_terminator: Sender<()>,
 }
 
 impl Multiplexer {
-    pub fn multiplex(ipc_recv: Receiver<String>) -> MultiplexResult {
+    pub fn multiplex<IpcReceiver, IpcSender>(ipc_send: IpcSender, ipc_recv: IpcReceiver) -> MultiplexResult
+    where
+        IpcReceiver: IpcRecv + 'static,
+        IpcSender: IpcSend + 'static, {
         let (request_send, request_recv) = channel::bounded(1);
         let (response_send, response_recv) = channel::bounded(1);
-        let (termination_send, termination_recv) = channel::bounded(0);
+        let receiver_terminator: Option<Mutex<Box<dyn Terminate>>> =
```

Comment:
Why should this be enclosed by `Mutex`?

Reply:
I copied the code without considering the reason. I found that it does not require the Mutex. I'll remove it. Thanks.

Reply:
`/// Here Mutex is used to make the Multiplxer Sync, while dyn Terminate isn't.` I think you will need this soon.